src/resources/dart-3-migration.md

```diff
@@ -0,0 +1,512 @@
+---
+title: Dart 3 migration guide
+description: Tips for how to migrate existing Dart code to be compatible with Dart 3.
+---
+
+Dart 3 is a new major release of Dart, that brings new core capabilities like
+records, patterns, and classes. Alongside these new capabilities the release
+also contains a number of changes, that may break existing code. The present
+document is intended to be a migration guide that may be helpful in resolving
+these.
+
+## Introduction
```

Comment:
Right under here, could we put how to actually get dart 3? Even if it's as simple as "run `dart pub upgrade`?

```suggestion
## Introduction

To get Dart 3 and be able to use its features, you need to run `dart pub upgrade`, and update the minimum SDK version in your [`pubspec.yaml`](/guides/packages#creating-a-pubspec): `sdk: ^3.0.0`

```

Reply:
Add above